crits/backdoors/handlers.py

```diff
@@ -0,0 +1,423 @@
+
+    if isinstance(source, basestring):
+        source = [create_embedded_source(source,
+                                         reference=source_reference,
+                                         method=source_method,
+                                         analyst=analyst)]
+    if not source:
+        retVal['message'] = "Missing source information."
+        return retval
+
+    # When creating a backdoor object we can potentially create multiple
+    # objects. If we are given a name but no version we will create an object
+    # with just the name (called the "family backdoor"). If given a name and a
+    # version we will create the family backdoor and the specific backdoor for
+    # that given version.
+
+    # In case we create more than one backdoor object, store the created ones
+    # in this list. The list is walked later on and attributes applied to each
+    # object.
+    objs = []
+
+    # First check if we have the family (name and no version).
+    family = Backdoor.objects(name=name, version='').first()
+
+    if not family:
+        family = Backdoor()
+        family.name = name
+        family.version = ''
+        # Family does not exist, new object. Details are handled later.
+        objs.append(family)
+    else:
+        # Family exists, old object. Add a new source only.
+        for s in source:
+            family.add_source(s)
+        family.save()
+        # In case this is the only object we create, make sure to return it.
+        retVal['object'] = family
+        retVal['id'] = str(family.id)
+        resp_url = reverse('crits.backdoors.views.backdoor_detail',
+                           args=[family.id])
+        retVal['message'] = 'Success: <a href="%s">%s</a>' % (resp_url,
+                                                              family.name)
+    # Now check if we have the specific instance for this name + version.
+    if version:
+        backdoor = Backdoor.objects(name=name, version=version).first()
+        if not backdoor:
+            backdoor = Backdoor()
+            backdoor.name = name
+            backdoor.version = version
+            # Backdoor does not exist, new object. Details are handled later.
+            objs.append(backdoor)
+        else:
+            # Backdoor exists, old object. Add a new source only.
+            for s in source:
+                backdoor.add_source(s)
+            backdoor.save()
+            resp_url = reverse('crits.backdoors.views.backdoor_detail',
+                               args=[backdoor.id])
+            retVal['message'] = 'Success: <a href="%s">%s</a>' % (resp_url,
+                                                                  backdoor.name)
+            retVal['object'] = family
+            retVal['id'] = str(family.id)
+
+    # At this point we have a family object and potentially a specific object.
+    # Add the common parameters to all objects in the list and save them.
+    for backdoor in objs:
```

Comment:
That is correct. If we have an existing family or specific instance backdoor we will not overwrite any of the things you mention (campaigns, tickets, buckets). My rationale for this is that I don't want to allow people to overwrite those things without explicitly acknowledging it by doing so on the backdoor details page.

You are correct that if you have an existing family backdoor and create a new specific instance backdoor that matches the family name a relationship will not be created. I will fix that now, good catch.